listener/upload.go

```diff
@@ -10,71 +10,83 @@ import (
 	"crypto/sha256"
 	"encoding/hex"
 	"encoding/json"
-	"fmt"
 	"github.com/RedHatInsights/patchman-clients/inventory"
 	"github.com/RedHatInsights/patchman-clients/vmaas"
 	"github.com/pkg/errors"
+	"github.com/segmentio/kafka-go"
 	"time"
 )
 
-func uploadHandler(event mqueue.PlatformEvent) {
-	tStart := time.Now()
-	defer utils.ObserveSecondsSince(tStart, messageHandlingDuration.WithLabelValues(EventUpload))
+type Host struct {
+	ID                    string                    `json:"id,omitempty"`
+	DisplayName           *string                   `json:"display_name,omitempty"`
+	AnsibleHost           *string                   `json:"ansible_host,omitempty"`
+	Account               string                    `json:"account,omitempty"`
+	InsightsID            string                    `json:"insights_id,omitempty"`
+	RhelMachineID         string                    `json:"rhel_machine_id,omitempty"`
+	SubscriptionManagerID string                    `json:"subscription_manager_id,omitempty"`
+	SatelliteID           string                    `json:"satellite_id,omitempty"`
+	FQDN                  string                    `json:"fqdn,omitempty"`
+	BiosUUID              string                    `json:"bios_uuid,omitempty"`
+	IPAddresses           []string                  `json:"ip_addresses,omitempty"`
+	MacAddresses          []string                  `json:"mac_addresses,omitempty"`
+	ExternalID            string                    `json:"external_id,omitempty"`
+	Created               string                    `json:"created,omitempty"`
+	Updated               string                    `json:"updated,omitempty"`
+	StaleTimestamp        *string                   `json:"stale_timestamp,omitempty"`
+	StaleWarningTimestamp *string                   `json:"stale_warning_timestamp,omitempty"`
+	CulledTimestamp       *string                   `json:"culled_timestamp,omitempty"`
+	Reporter              string                    `json:"reporter,omitempty"`
+	Tags                  []inventory.StructuredTag `json:"tags,omitempty"`
+	SystemProfile         inventory.SystemProfileIn `json:"system_profile,omitempty"`
+}
 
-	if event.B64Identity == nil {
-		utils.Log("inventoryID", event.ID).Error("Identity not provided")
-		messagesReceivedCnt.WithLabelValues(EventUpload, ReceivedErrorIdentity).Inc()
-		return
-	}
+type HostEgressEvent struct {
+	Type             string                 `json:"type"`
+	PlatformMetadata map[string]interface{} `json:"platform_metadata"`
```

Comment:
Do we need this field?

Reply:
Not ATM, but it should contain identity and so on, I'd rather leave it here, for logging purposes at least.